src/bidiMapper/domains/script/WindowRealm.ts

```diff
@@ -0,0 +1,205 @@
+import type {Protocol} from 'devtools-protocol';
+
+import type {CdpClient} from '../../../cdp/CdpClient.js';
+import {
+  NoSuchNodeException,
+  type Script,
+  UnknownErrorException,
+  type BrowsingContext,
+} from '../../../protocol/protocol.js';
+import {CdpErrorConstants} from '../../../utils/CdpErrorConstants.js';
+import type {LoggerFn} from '../../../utils/log.js';
+import type {BrowsingContextImpl} from '../context/BrowsingContextImpl.js';
+import type {BrowsingContextStorage} from '../context/BrowsingContextStorage.js';
+import type {EventManager} from '../session/EventManager.js';
+
+import {Realm} from './Realm.js';
+import type {RealmStorage} from './RealmStorage.js';
+import {SharedIdParser} from './SharedIdParser.js';
+
+export class WindowRealm extends Realm {
+  readonly #browsingContextId: BrowsingContext.BrowsingContext;
+  readonly #browsingContextStorage: BrowsingContextStorage;
+  readonly #sharedIdWithFrame: boolean;
+  readonly sandbox: string | undefined;
+
+  constructor(
+    browsingContextId: BrowsingContext.BrowsingContext,
+    browsingContextStorage: BrowsingContextStorage,
+    cdpClient: CdpClient,
+    eventManager: EventManager,
+    executionContextId: Protocol.Runtime.ExecutionContextId,
+    logger: LoggerFn | undefined,
+    origin: string,
+    realmId: Script.Realm,
+    realmStorage: RealmStorage,
+    sandbox: string | undefined,
+    sharedIdWithFrame: boolean
+  ) {
+    super(
+      cdpClient,
+      eventManager,
+      executionContextId,
+      logger,
+      origin,
+      realmId,
+      realmStorage
+    );
+
+    this.#browsingContextId = browsingContextId;
+    this.#browsingContextStorage = browsingContextStorage;
+    this.#sharedIdWithFrame = sharedIdWithFrame;
+    this.sandbox = sandbox;
+
+    this.initialize();
+  }
+
+  #getBrowsingContextId(navigableId: string): string {
+    const maybeBrowsingContext = this.#browsingContextStorage
+      .getAllContexts()
+      .find((context) => context.navigableId === navigableId);
+    return maybeBrowsingContext?.id ?? 'UNKNOWN';
+  }
+
+  get browsingContext(): BrowsingContextImpl {
+    return this.#browsingContextStorage.getContext(this.#browsingContextId);
+  }
+
+  override get indirectBrowsingContexts(): [BrowsingContextImpl] {
```

Comment:
It can be a bit confusing in case of window realm, where it's direct. `associatedBrowsingContexts` was more clear. Consider rename back.

Reply:
Sure.